ffi/src/repr_c.rs

```diff
@@ -195,10 +195,10 @@ impl<'slice, R: Ir<Type = S> + CTypeConvert<'slice, S, C> + Clone, S: NonTransmu
     }
 }
 
-impl<R: Ir<Type = S> + CType<S>, S: NonTransmute> CType<Vec<S>> for Vec<R> {
```

Comment:
Because the implementation of `FfiType` requires that both `Ir` and `Ctype` are implemented for some `T`. Because of this `Ir` and `Ctype` don't have to be mutually dependent. This looks simpler while achieving the same

So even if there is no explicit relationship between `Ir` and `CType` they both have to be implemented for a type to derive implementation of `FfiType`